src/unittest/distance.cpp

```diff
@@ -0,0 +1,1748 @@
+            TestDistanceIndex::SnarlDistances& sd1 = di.snarlIndex.at(
+                                    make_pair(snarl1->start().node_id(),
+                                              snarl1->start().backward()));
+
+            const Snarl* snarl2 = snarl_manager.into_which_snarl(2, false);
+            TestDistanceIndex::SnarlDistances& sd2 = di.snarlIndex.at(
+                            make_pair(snarl2->start().node_id(),
+                                     snarl2->start().backward()));
+
+
+            REQUIRE(sd1.snarlDistance(make_pair(1, false), make_pair(7, false)) == 3);
+            REQUIRE(sd1.snarlDistance(make_pair(1, false), make_pair(7, false)) == 3);
+            REQUIRE(sd1.snarlDistance(make_pair(
+                       snarl2->start().node_id(), !snarl2->start().backward()), 
+                      make_pair(snarl2->start().node_id(), 
+                                           snarl2->start().backward())) == -1);
+            REQUIRE(sd2.snarlDistance(make_pair(4, true), make_pair(4, false)) == 6);
+            REQUIRE(sd1.snarlDistance(make_pair(5, true), make_pair(5, false)) == 13);
+            REQUIRE(sd1.snarlDistance(make_pair(5, true), make_pair(7, false)) == 14);
+            REQUIRE(sd1.snarlDistance(make_pair(7, true), make_pair(7, false)) == 13);
+
+        }
+
+
+        SECTION ("Distance functions") {
+            const Snarl* snarl1 = snarl_manager.into_which_snarl(1, false);
+            const Snarl* snarl2 = snarl_manager.into_which_snarl(2, false);
+            const Snarl* snarl3 = snarl_manager.into_which_snarl(3, false);
+
+            pos_t pos1 = make_pos_t(1, false, 0);
+            pos_t pos2 = make_pos_t(2, false, 0);
+            pos_t pos3 = make_pos_t(3, false, 0);
+            pos_t pos4 = make_pos_t(4, false, 0);
+            pos_t pos5 = make_pos_t(5, false, 0);
+            pos_t pos6 = make_pos_t(6, false, 0);
+            pos_t pos7 = make_pos_t(7, false, 0);
+
+            REQUIRE(di.distance(snarl1, snarl2, pos1, pos4) == 5);
+            REQUIRE(di.distance(snarl1, snarl1, pos5, pos6) == 12);
+            REQUIRE(di.distance(snarl2, snarl1, pos2, pos7) == 7);
+
+            REQUIRE(distance(&graph, pos1, pos4) == 5);
+            REQUIRE(distance(&graph, pos5, pos6) == 12);
+            REQUIRE(distance(&graph, pos2, pos7) == 7);
+        }
+    }//End test case
+    TEST_CASE("Nested unary snarls", "[dist]") {
+        VG graph;
+        Node* n1 = graph.create_node("GCA");
+        Node* n2 = graph.create_node("T");
+        Node* n3 = graph.create_node("G");
+        Node* n4 = graph.create_node("CTGA");
+        Node* n5 = graph.create_node("GCA");
+        Node* n6 = graph.create_node("T");
+        Node* n7 = graph.create_node("G");
+        Node* n8 = graph.create_node("CTGA");
+
+        Edge* e1 = graph.create_edge(n1, n2);
+        Edge* e2 = graph.create_edge(n1, n4);
+        Edge* e3 = graph.create_edge(n1, n8);
+        Edge* e4 = graph.create_e
```

Comment:
It's probably fine, but for the record `rand()` is considered a pretty poor RNG these days. It would probably be better to use something like `uniform_int_distribution<int>(0, node1->sequence().size())(generator)`, or for the `bool`s below, `uniform_int_distribution<int>(0, 1)(generator)`